content/docs/user-guide/concepts/data-pipelines.md

```diff
@@ -0,0 +1,34 @@
+---
+name: 'Data Pipelines'
+match: ['data pipeline', 'pipeline', 'pipelines']
+tooltip: >-
+  A data pipeline is a series of data processing stages, chained by their
+  outputs and inputs. They use some initial data, may produce intermediate
+  artifacts, and reach a final result.
+  [📖](/doc/user-guide/concepts/data-pipelines)
+description: >-
+  A data pipeline is a series of data processing stages. They use some initial
+  data, may produce intermediate artifacts, and reach a final result.
+---
+
+<!-- keywords: data pipeline, machine learning pipeline, devops for data science, devops for machine learning, "MLops", "what is a data pipeline?", "data pipeline examples", "machine learning (ML) workflow", "data science workflow", "data science pipeline workflow" -->
+
+# Data Pipelines
+
+A data pipeline, in general, is a series of data processing <abbr>stages</abbr>
+like data gathering, data transformation, model training, model testing and
+reporting <abbr>metrics</abbr> and results. A pipeline may produce intermediate
```

Comment:
As you may have realized, a lot of the text in this branch comes from existing content, in this case https://dvc.org/doc/command-reference/dag

So the PR should also remove most of those explanations from the cmd ref. — another reason to focus on just one concept at a time per PR probably — not to mention all the link/tooltip updates that some of these concepts will require throughout docs 😱

Reply:
I'm not against using duplicate material as soon as it fits to the context. Concepts pages should be revised to contain less details but overall removing material from other parts of the documentation may not be necessary. 

Users will be reading these sporadically and they probably will forget what they read in concepts when they are reading the command ref. Aesthetically it may not be pleasing to repeat ourselves but from POV of a user, these are separate instances of reading. Also we can't be sure the user will ever read the other if they read one of them.

Reply:
> removing material from other parts of the documentation may not be necessary

That's actually an important goal for this task (see https://github.com/iterative/dvc.org/issues/550#issuecomment-673585050). Some (a little) redundancy is OK but there should be a single source of full/comprehensive explanation for these concepts. Keep in mind that maintaining duplicated content is hard and error-prone 😅 (just like code)

> Users will be reading these sporadically and they probably will forget

TBH I don't expect users to navigate to concepts much it at all (from nav). Like you suggested without context they're pretty hard to care about or understand. They'll probably only open them from links we place in other docs, and from the tooltips of course.

Reply:
It would be great though for this section to serve as an entry point. May be index page can give an overview of the most important ones? how do hey relate to each other.

Reply:
Good idea. I agree about an guiding index page for those who do navigate in here. BUT probably the 1st PR (after this one) will only introduce 1 or a few concepts and the index may not make sense until we have most of them up.